extensions/amp-story-interactive/1.0/amp-story-interactive-abstract.js

```diff
@@ -18,13 +18,16 @@ import {
   ANALYTICS_TAG_NAME,
   StoryAnalyticsEvent,
   getAnalyticsService,
-} from './story-analytics';
+} from '../../amp-story/1.0/story-analytics';
 import {
   Action,
   StateProperty,
   getStoreService,
-} from './amp-story-store-service';
-import {AnalyticsVariable, getVariableService} from './variable-service';
+} from '../../amp-story/1.0/amp-story-store-service';
```

Comment:
I'm afraid the `getStoreService` method here will import the whole store service, duplicating the code between amp-story.js and amp-story-interactive.js.
Can you check in the build if that's the case? If yes, you can get rid of this duplicated code and save a lot of bytes by using AMP's dependency injection system, and use the AMP service getter. It'll get the instance from amp-story.js without importing the whole class in amp-story-interactive.js. We do it here:

https://github.com/ampproject/amphtml/blob/d788985c7e7c7cd31df3268474fb3536a5144287/extensions/amp-story-education/0.1/amp-story-education.js#L88-L91

Can you check if there's a way to do that with the other services you're using? If the whole point is to save bytes, let's try to reduce code duplication, we might actually be making things worse if we're not careful :))

Reply:
Since the interactive extension might load before the amp-story, the StoreService code needs to get loaded with this extension. Unless we can guarantee that the interactive extension loads after the other one, we need to use the getStoreService.

Any ideas on how to do this?

Reply:
Yes, let's use the async service getter: https://github.com/ampproject/amphtml/blob/bf5be9f633e256ae107ee0e50fd783a1a8d40dfa/src/services.js#L479-L487

I pinged you a code example offline if needed.

Reply:
Done with the ones that I could, some services didn't have the Promise so I had to call them directly

Reply:
Thank you, I filed https://github.com/ampproject/amphtml/issues/29282 for us to investigate as a follow up.